lib/src/working_copy.rs

```diff
@@ -489,6 +509,13 @@ impl TreeState {
                     continue;
                 }
                 let sub_path = dir.join(&RepoPathComponent::from(name));
+                if let Some(file_state) = self.file_states.get(&sub_path) {
+                    if matches!(file_state.file_type, FileType::GitSubmodule{ .. }) {
```

Comment:
If you remove the `id` member in `FileType::GitSubmodule`, then this `matches!()` can be a regular comparison (`if file_state.file_type == FileType::GitSubmodule`).